src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -235,12 +235,13 @@ export const Map: React.FC<MapProps> = ({
           src={venue.mapBackgroundImageUrl ?? DEFAULT_MAP_BACKGROUND}
           alt=""
         />
-
-        <div className="party-map-grid-container" style={gridContainerStyles}>
-          {mapGrid}
-          {partygoersOverlay}
-          {roomOverlay}
-        </div>
+        {totalRows && (
```

Comment:
@soichih Can you please explain the intent behind your logic here please. What was the underlying issue as you see it (at a technical level), and why does this change resolve it?

Reply:
totalRows is initialized to be 0, so this prevents rooms to be drawn until it is reset to non-0 (truthy) value thus eliminating the flicker.

Reply:
That makes sense.

I wonder if there is a better way we can fix/improve this 'higher up'?

eg. (musing, these may not necessarily be good ideas)

- initialising with a better default value
- improving the `useEffect` that loads the image just to calculate it's width/height ratio
  - https://github.com/sparkletown/sparkle/blob/staging/src/components/templates/PartyMap/components/Map/Map.tsx#L71-L85
  - Potentially could use `useLayoutEffect` here instead? Though might block rendering too long while loading image?
    - https://reactjs.org/docs/hooks-reference.html#uselayouteffect
- Maybe just treating the whole map as 'loading' until we have the correct `totalRows` value calculated
- etc

---

@mike-lvov What do you think about the solution in this PR + the potential ideas I listed above + any other solutions you can think of for this?

Reply:
Anyway, {totalRows &} will result in displaying `0` on the Map, which we want to avoid. moving it to `hasRows` const and using that value to display loading is a better approach IMO

Reply:
How about `{(totalRows > 0) & ... }`?  You should avoid creating unnecessary variables / states, etc.

Reply:
> You should avoid creating unnecessary variables / states, etc.

Extracting a const with a semantic name has negligible impact on performance/etc, while increasing the semantic readability of the code, so it’s a pattern we often follow.

Reply:
> Anyway, {totalRows &} will result in displaying 0 on the Map, which we want to avoid

@mike-lvov Will it? 0 is a falsy value, so I believe it wouldn’t render and be fine (though it’s not as explicit/obvious code at first ‘read’ as we could make it)

Reply:
> Will it? 0 is a falsy value, so I believe it wouldn’t render and be fine

It will

Reply:
- https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator
  - > Note that returning a falsy expression will still cause the element after && to be skipped but will return the falsy expression. In the example below, `<div>0</div>` will be returned by the render method.

@mike-lvov Seems you're correct.. I didn't realise it worked that way here.. weird implementation choice on React's side of things.

Reply:
> How about {(totalRows > 0) & ... }? You should avoid creating unnecessary variables / states, etc.

I like this idea, if you move that to a const as per 
> Extracting a const with a semantic name has negligible impact on performance/etc, while increasing the semantic readability of the code, so it’s a pattern we often follow

Reply:
- Reverted back to `room.title`. Gonna make a separate PR where will try to change all occurrences of `index` as a key to `key={`${room.title}-${index}`}` as @0xdevalias suggested.
- extracted `const hasRows = totalRows > 0`
in https://github.com/sparkletown/sparkle/pull/1403/commits/b5dc99e44573b1e11404f11e41121f3105e299c7